### PR TITLE
Create Felvin FAQ instant app

### PR DESCRIPTION
**Pull request by hargup**
Screenshot 2022-02-27 at 2 30 15 AM

Answers basic questions about Felvin.

### CODE REVIEW

File: apps/felvin-faq/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// This where you can process the query and try to convert it into some meaningful data.
+async function queryToData({
+  query,
+}) {
+  const sheetID = "1duJLsNW6IYmOnR9JItUuEHtlxS79ce4LalzQcHfKvCA";
+  const response = await fetch(`https://low-code-service.felvin.com/api/?sheetID=${sheetID}`)
+  const data = await response.json()
+  for(const item of data){
+    console.log(item)
+    if(item["Question"] === query) {
```
Comment (sahil-shubham) [inline apps/felvin-faq/src/App.jsx:55]: Not too sure about this way of string matching. Had to see the json of the whole sheet to know which questions to ask and whether or not punctuations are needed.

File: apps/felvin-faq/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { Component, queryToData } from "./App";
+
+const App: InstantApp = {
+  id: "@felvin-community/felvin-faq",
+  name: "Felvin FAQ",
+  description: "Shows answers to common Felvin FAQ questions",
+  queryToData,
+  Component,
+  // screenshotPath: "./files/screenshot.png",
+  // exampleSearchQueries: [],
```
Comment (sahil-shubham) [inline apps/felvin-faq/src/index.ts:11]: Let's add these, would have to add all the questions as example search queries.

**Review by sahil-shubham — APPROVED** [review]
Looks good, easy way to beat all the SEO stuff regarding the company and answer questions directly. 

Left a few comments, let me know if you want me to make the changes.

File: apps/felvin-faq/src/App.jsx (hunk 1 above)
Comment (hargup) [reply apps/felvin-faq/src/App.jsx:55]: I should probably normalize the query and the question string. Little later, will implement more robust matchers.

File: apps/felvin-faq/src/index.ts (hunk 2 above)
Comment (hargup) [reply apps/felvin-faq/src/index.ts:11]: Thanks for pointing this out.

File: apps/felvin-faq/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  return string1 === string2;
+}
+
+const siftMatcher = (string1, string2) => {
+  const N = Math.max(string1.length, string2.length)
+  return sift4(string1, string2)/(N*1.0)
+}
+
+// Default matcher
+const matchStrings = (string1, string2, threshold=0.8, preprocessor=fingerprint, matcher=identityMatcher) => {
```
Comment (hargup) [inline apps/felvin-faq/src/App.jsx:23]: @N-Shar-ma @sahil-shubham Here's my attempt at creating smarter matchers. See if these are working for you locally.

File: apps/felvin-faq/src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+// ------------- Configs ---------------------------
+const exactMatchConfig = {threshold: 1, preprocessor: identity, matcher: identityMatcher}
+const typoTolerantConfig = {threshold: 0.8, preprocessor: fingerprint, matcher: siftMatcher}
+// There can be more configs
+const defaultConfig = typoTolerantConfig
+
+
+// ------------- Function to use -------------------
+
+const matchStrings = (string1, string2, config=defaultConfig) => {
```
Comment (hargup) [inline apps/felvin-faq/src/App.jsx:70]: @N-Shar-ma @sahil-shubham Please see if this system makes sense to you.

Comment (hargup) [reply apps/felvin-faq/src/App.jsx:70]: If it doesn't, we'll move the functions to felvin core

**Comment by hargup** [comment]
Closing for lack of activity.

**Pull request closed without merging**